sys/include/senml.h

```diff
@@ -0,0 +1,377 @@
+} senml_unit_t;
+
+/**
+ * @brief SenML numeric value types.
+ *
+ */
+typedef enum {
+    SENML_TYPE_NUMERIC_UINT,    /**< Unsigned integer */
+    SENML_TYPE_NUMERIC_INT,     /**< Integer */
+    SENML_TYPE_NUMERIC_FLOAT,   /**< Floating point number */
+    SENML_TYPE_NUMERIC_DOUBLE,  /**< Double-precision floating point number */
+    SENML_TYPE_NUMERIC_DECFRAC, /**< Decimal fraction */
+} senml_value_type_t;
+
+/**
+ * @brief Decimal fraction containing a value in the form of m * 10^e.
+ */
+typedef struct {
+    int32_t e;  /**< Exponent */
+    int32_t m;  /**< Mantissa */
+} senml_decfrac_t;
+
+/**
+ * @brief SenML numeric value.
+ *
+ * Various SenML attributes (see @ref senml_attr_t) may contain any 'numeric'
+ * types. This struct is used to contain these.
+ */
+typedef struct {
+    senml_value_type_t type; /**< Type of the value */
+    union {
+        uint64_t u;
+        int64_t i;
+        float f;
+        double d;
+        struct { int32_t e; int32_t m; } df /** Decimal fraction */;
+    } value; /**< Value data */
+} senml_numeric_t;
+
+/**
+ * @brief SenML common record attributes.
+ * All of these values are optional: empty or 0 values will not be encoded.
+ */
+typedef struct {
+    const char *base_name;          /**< Base Name */
+    senml_numeric_t base_time;      /**< Base Time */
+    senml_unit_t base_unit;         /**< Base Unit */
+    senml_numeric_t base_value;     /**< Base Value */
+    senml_numeric_t base_sum;       /**< Base Sum */
+    uint64_t base_version;          /**< Base Version */
+    const char *name;               /**< Name of the measurement */
+    senml_unit_t unit;              /**< Unit */
+    senml_numeric_t sum;            /**< Sum */
+    senml_numeric_t time;           /**< Time of the measurement (relative or Unix) in seconds */
+    senml_numeric_t update_time;    /**< Maximum time before the next sensor value */
+} senml_attr_t;
+
+/**
+ * @brief SenML string value.
+ */
+typedef struct {
+    senml_attr_t attr;      /**< SenML attributes */
+    senml_numeric_t value;  /**< Value */
+} senml_value_t;
+
+/**
+ * @brief SenML string value.
+ */
+typedef struct {
+    senml_attr_t attr;  /**< SenML attributes */
+    const char *value;  /**< Value */
+    size_t len;         /**< Value length */
+} senml_string_value_t;
+
+/**
+ * @brief SenML boolean value.
+ */
+typedef struct {
+    senml_attr_t attr;  /**< SenML attributes */
+    bool value;         /**< Value */
+} senml_bool_value_t;
+
+/**
+ * @brief SenML data value.
+ */
+typedef struct {
+    senml_attr_t attr;      /**< SenML attributes */
+    const uint8_t *value;   /**< Value */
+    size_t len;             /**< Value length */
+} senml_data_value_t;
+
+/**
+ * @brief Create a floating point numeric value.
+ *
+ * @param v Value to encode.
+ * @return Numeric value containing the given value.
+ */
+inline senml_numeric_t senml_float(float v)
```

Comment:
```suggestion
static inline senml_numeric_t senml_float(float v)
```

And for other cases below.